### PR TITLE
disko-install: make output deterministic

### LINKED ISSUE #827: non-deterministic disko-install
https://github.com/nix-community/disko/blob/bdbdb725d632863bdedb80baabf21327614dd237/disko-install#L193

this makes disko-install non-deterministic, so it always has to build the derivation

I've been trying to use it on a 2GB ram SBC, and it gets oom-killed

calling the config's `diskoScript`, and followed by `nixos-install ...` seems to work fine

my final goal is to use disko-install on low-ram SBC's, so if there's something else I missed please let me know

### PR DESCRIPTION
**Pull request by iFreilicht**
Fixes #827

@matdibu FYI

### CODE REVIEW

**Comment by iFreilicht** [comment]
Alright I had time to properly test this now, and it works!

On master:

```console
# sudo nix run '.#disko-install' -- --dry-run --disk main /dev/asdf --flake '.#testmachine'
these 2 derivations will be built:
  /nix/store/1dxaxs72mgm3f989225hgs9w0l209cwm-disko.drv
  /nix/store/iyjdzd298y7hh2vdgn9nrqlrknzqiy64-disko.drv
building '/nix/store/1dxaxs72mgm3f989225hgs9w0l209cwm-disko.drv'...
building '/nix/store/iyjdzd298y7hh2vdgn9nrqlrknzqiy64-disko.drv'...
Would run: /nix/store/kgmcj274zh83yb7gwadvymsv69zzkhl5-disko
Would run: nixos-install --system '/nix/store/b20shmmwlg1np3hflxjklm58a77gsnrs-nixos-system-disko-machine-24.11.20240901.b833ff0' --root '/tmp/tmp.1soo4VO6rU'
# sudo nix run '.#disko-install' -- --dry-run --disk main /dev/asdf --flake '.#testmachine'
these 2 derivations will be built:
  /nix/store/456549r013s5dpa7fpjvlbgk3dyd465d-disko.drv
  /nix/store/i1xfmwrj4b74x9gwryparyq4x0a954gm-disko.drv
building '/nix/store/456549r013s5dpa7fpjvlbgk3dyd465d-disko.drv'...
building /nix/store/i1xfmwrj4b74x9gwryparyq4x0a954gm-disko.drv'...
Would run: /nix/store/0ma5r0swjwc4xsgx4s06pi9q8s6241y6-disko
Would run: nixos-install --system '/nix/store/b20shmmwlg1np3hflxjklm58a77gsnrs-nixos-system-disko-machine-24.11.20240901.b833ff0' --root '/tmp/tmp.SGnbALeYr4'
```

On this branch:

```console
# sudo nix run '.#disko-install' -- --dry-run --disk main /dev/asdf --flake '.#testmachine'
these 2 derivations will be built:
  /nix/store/j185c9vxblajx27biygfpw5kqk0lzlfa-disko.drv
  /nix/store/q7wf7y4xqahvrifwyssxmsy6v64rkk49-disko.drv
building '/nix/store/j185c9vxblajx27biygfpw5kqk0lzlfa-disko.drv'...
building '/nix/store/q7wf7y4xqahvrifwyssxmsy6v64rkk49-disko.drv'...
Would run: /nix/store/vj7mdgcki9l4wdnfyp4dfswns2hlk5nd-disko
Would run: nixos-install --system '/nix/store/b20shmmwlg1np3hflxjklm58a77gsnrs-nixos-system-disko-machine-24.11.20240901.b833ff0' --root '/mnt/disko-install-root'
# sudo nix run '.#disko-install' -- --dry-run --disk main /dev/asdf --flake '.#testmachine'
Would run: /nix/store/vj7mdgcki9l4wdnfyp4dfswns2hlk5nd-disko
Would run: nixos-install --system '/nix/store/b20shmmwlg1np3hflxjklm58a77gsnrs-nixos-system-disko-machine-24.11.20240901.b833ff0' --root '/mnt/disko-install-root'
```

**Review by matdibu — APPROVED** [review]
thank you!

File: disko-install [hunk 1]
```diff
@@ -190,7 +190,8 @@ main() {
     exit 1
   fi
 
-  mountPoint=$(mktemp -d)
+  mountPoint="/mnt/disko-install-root"
+  mkdir -p "${mountPoint}"
   chmod 755 "${mountPoint}" # bchachefs wants 755
```
Comment (Enzime) [inline disko-install:195]:
```suggestion
  chmod 755 "${mountPoint}" # bcachefs wants 755
```

**Review by Enzime — CHANGES_REQUESTED** [review]
LGTM otherwise

File: disko-install (hunk 1 above)
Comment (iFreilicht) [reply disko-install]: Ah yeah good catch, fixed it!

**Review by Enzime — APPROVED** [review]

**Pull request merged**